array/binary.go

```diff
@@ -15,42 +15,60 @@ func And(l, r *Boolean, mem memory.Allocator) (*Boolean, error) {
 	b := NewBooleanBuilder(mem)
 	b.Resize(n)
 	for i := 0; i < n; i++ {
-		if l.IsValid(i) && r.IsValid(i) {
-			b.Append(l.Value(i) && r.Value(i))
-		} else {
-			b.AppendNull()
+		var elmL *bool
+		if l.IsValid(i) {
+			x := l.Value(i)
+			elmL = &x
 		}
-	}
-	a := b.NewBooleanArray()
-	b.Release()
-	return a, nil
-}
-
-func AndLConst(l bool, r *Boolean, mem memory.Allocator) (*Boolean, error) {
-	n := r.Len()
-	b := NewBooleanBuilder(mem)
-	b.Resize(n)
-	for i := 0; i < n; i++ {
+		var elmR *bool
 		if r.IsValid(i) {
-			b.Append(l && r.Value(i))
-		} else {
+			x := r.Value(i)
+			elmR = &x
+		}
+
+		if elmL == nil && elmR == nil {
+			// both sides are null
 			b.AppendNull()
+		} else if elmL == nil || elmR == nil {
+			// one side is null, the other is false
+			if (elmL != nil && !*elmL) || (elmR != nil && !*elmR) {
+				b.Append(false)
+			} else {
+				b.AppendNull()
+			}
+		} else {
+			// no nulls
+			b.Append(*elmL && *elmR)
 		}
 	}
 	a := b.NewBooleanArray()
 	b.Release()
 	return a, nil
 }
 
-func AndRConst(l *Boolean, r bool, mem memory.Allocator) (*Boolean, error) {
-	n := l.Len()
+func AndConst(fixed *bool, arr *Boolean, mem memory.Allocator) (*Boolean, error) {
```

Comment:
When I added the initial "vec repeat" support for logical ops I didn't realize the op was commutative. Now that I'm wise to this, I updated this to just swap the args around at the call sites instead of having L and R versions.